source/state_representation/include/state_representation/Space/Cartesian/CartesianWrench.hpp

```diff
@@ -7,205 +7,243 @@
+  CartesianWrench(const CartesianWrench& wrench);
+
+  /**
+   * @brief Copy constructor from a CartesianState
+   */
+  CartesianWrench(const CartesianState& state);
+
+  /**
+   * @brief Construct a CartesianWrench from a force given as a vector of coordinates.
+   */
+  explicit CartesianWrench(const std::string& name, const Eigen::Vector3d& force, const std::string& reference = "world");
+
+  /**
+   * @brief Construct a CartesianWrench from a force given as a vector of coordinates and a quaternion.
+   */
+  explicit CartesianWrench(const std::string& name, const Eigen::Vector3d& force, const Eigen::Vector3d& torque, const std::string& reference = "world");
+
+  /**
+   * @brief Construct a CartesianWrench from a single 6d wrench vector
+   */
+  explicit CartesianWrench(const std::string& name, const Eigen::Matrix<double, 6, 1>& wrench, const std::string& reference = "world");
+
+  /**
+   * @brief Constructor for the zero wrench
+   * @param name the name of the state
+   * @param the name of the reference frame
+   * @return CartesianWrench with zero values
+   */
+  static const CartesianWrench Zero(const std::string& name, const std::string& reference = "world");
+
+  /**
+   * @brief Constructor for a random wrench
+   * @param name the name of the state
+   * @param the name of the reference frame
+   * @return CartesianWrench random wrench
+   */
+  static const CartesianWrench Random(const std::string& name, const std::string& reference = "world");
+
+  /**
+   * @brief Copy assignement operator that have to be defined to the custom assignement operator
+   * @param pose the pose with value to assign
+   * @return reference to the current pose with new values
+   */
+  CartesianWrench& operator=(const CartesianWrench& pose);
+
+  /**
+   * @brief Overload the = operator from a CartesianState
+   * @param state CartesianState to get the wrench from
+   */
+  CartesianWrench& operator=(const CartesianState& state);
+
+  /**
+   * @brief Set the values of the 6D wrench from a 6D Eigen Vector
+   * @param wrench the wrench as an Eigen Vector
+   */
+  CartesianWrench& operator=(const Eigen::Matrix<double, 6, 1>& wrench);
+
+  /**
+   * @brief Overload the *= operator
+   * @param wrench CartesianWrench to multiply with
+   * @return the current CartesianWrench multiplied by the CartesianWrench given in argument
+   */
+  CartesianWrench& operator*=(const CartesianWrench& twist);
+
+  /**
+   * @brief Overload the * operator with a wrench
+   * @param wrench CartesianWrench to multiply with
+   * @return the current CartesianWrench multiplied by the CartesianWrench given in argument
+   */
+  const CartesianWrench operator*(const CartesianWrench& twist) const;
+
+  /**
+   * @brief Overload the * operator with a state
+   * @param state CartesianState to multiply with
+   * @return the current CartesianWrench multiplied by the CartesianState given in argument
+   */
+  const CartesianState operator*(const CartesianState& state) const;
```

Comment:
For the same reason as the twist, I'm not sure if it makes sense to have a state = wrench*state without any scaling property (in this case inertia, in the case of twist it's time). But, we also had a discussion a little while ago about the place of wrench in state representation. Overall this question is one for a future PR.